src/plugins/data/server/search/opensearch_search/decide_client.ts

```diff
@@ -9,14 +9,17 @@ import { IOpenSearchSearchRequest } from '..';
 export const decideClient = async (
   context: RequestHandlerContext,
   request: IOpenSearchSearchRequest,
+  withDataSourceEnabled: boolean = false,
   withLongNumeralsSupport: boolean = false
 ): Promise<OpenSearchClient> => {
-  // if data source feature is disabled, return default opensearch client of current user
-  const client =
+  const defaultOpenSearchClient = withLongNumeralsSupport
+    ? context.core.opensearch.client.asCurrentUserWithLongNumeralsSupport
+    : context.core.opensearch.client.asCurrentUser;
+
+  const dataSourceClient =
     request.dataSourceId && context.dataSource
       ? await context.dataSource.opensearch.getClient(request.dataSourceId)
-      : withLongNumeralsSupport
-      ? context.core.opensearch.client.asCurrentUserWithLongNumeralsSupport
-      : context.core.opensearch.client.asCurrentUser;
-  return client;
+      : defaultOpenSearchClient;
```

Comment:
If default cluster is true while data source plugin is enabled, we should use the default opensearch client, right?

Reply:
It depends on dataSourceId.

If DataSourceId not exist or empty, we will use default opensearch client.
If DataSourceId exist, we will try to get MD client to sever that DataSourceId. And 404 might be returned if dataSourceId is not correct.

Reply:
Got it!